Extensions/retags.js

```diff
@@ -19,111 +19,157 @@ XKit.extensions.retags = {
+			} else if ($element.hasClass('ui_note')) {
+				if ($element.find('.part_response').length) {
+					$element.addClass('is_response');
 				}
-				cls = 'is_retags';
-				$c = $t.find('.stage');
-				url = $c.find('.part_glass').attr('href');
+				retagClass = 'is_retags';
+				$target = $element;
+				url = $target.find('.part_glass').attr('href');
 			// dashboard
-			} else if ($t.hasClass('notification') && !XKit.browser().mobile) {
-				$c = $t.find('.notification_sentence');
-				url = $c.find('.notification_target').attr('href');
+			} else if ($element.hasClass('notification') && !XKit.browser().mobile) {
+				$target = $element.find('.notification_sentence');
+				url = $target.find('.notification_target').attr('href');
 			// mobile
-			} else if ($t.hasClass('notification') && XKit.browser().mobile) {
-				cls = 'is_retags';
-				$c = $t.find('.notification-wrapper');
-				url = $c.find('a').not('.notification-username').attr('href');
-			}
-			//we don't need to put tags on a reply, but we also don't need to hide it
-			if ($t.hasClass('is_reply') || $t.hasClass('is_answer') || $t.hasClass('type_6')) {
-				return;
+			} else if ($element.hasClass('notification') && XKit.browser().mobile) {
+				retagClass = 'is_retags';
+				$target = $element.find('.notification-wrapper');
+				url = $target.find('a').not('.notification-username').attr('href');
 			}
 			if (url) {
 				url = url.split('/');
-				var host = url[2], id = url[4], cache = 'post_'+id;
-				if (XKit.storage.get('retags', cache, null) !== null) {
-					XKit.extensions.retags.append($t, cls, $c, XKit.storage.get('retags', cache));
-				} else {
-					XKit.extensions.retags.request($t,cls,$c,cache,'https://api.tumblr.com/v2/blog/'+host+'/posts/info?id='+id+'&api_key='+XKit.extensions.retags.api_key);
-				}
+				var host = url[2], id = url[4];
+				XKit.extensions.retags.request($element, retagClass, host, id)
+					.then(function(retags) {
+						$target.append(retags);
+					});
 			}
 		});
 	},
 
-	request: function($t,cls,$c,cache,url) {
-		GM_xmlhttpRequest({
-			method: 'GET',
-			url: url,
-			onload: function(data){
-				var tags = JSON.parse(data.responseText).response.posts[0].tags;
-				if (XKit.extensions.retags.count_cached_posts() > XKit.extensions.retags.POST_CACHE_CLEAR_THRESHOLD || XKit.storage.quota('retags') < 100) {
+	request: function($element, retagClass, host, id) {
+		var url = 'https://api.tumblr.com/v2/blog/' + host + '/posts/info?id=' + id + '&api_key=' + XKit.extensions.retags.api_key;
+		var deferred = $.Deferred();
+		var cache_label = 'post_' + id;
+		//if it's already cached, get from cache
+		if (XKit.storage.get('retags', cache_label, null) !== null) {
+			var retags = XKit.extensions.retags.build_tag_collection($element, retagClass, XKit.storage.get('retags', cache_label));
+			deferred.resolve(retags);
+			return deferred;
+		}
+		//otherwise make API call
+		var tags = [];
+		$.get(url)
+			.done(function(data){
```

Comment:
use the standard promise API of .then and .catch rather then the old jquery specific API